src/test/java/org/opensearch/neuralsearch/common/BaseNeuralSearchIT.java

```diff
@@ -623,6 +626,28 @@ protected void deleteModel(String modelId) {
         );
     }
 
+    protected void pollForModelState(String modelId, Set<MLModelState> exitModelStates, int intervalMs, int maxAttempts)
+        throws InterruptedException {
+        MLModelState currentState = null;
```

Comment:
can we move this initialization into the loop, it's not required to have it in a top level scope